src/Kralizek.Extensions.Configuration.AWSSecretsManager/Internal/SecretsManagerConfigurationProviderOptions.cs

```diff
@@ -62,13 +61,24 @@ public class SecretsManagerConfigurationProviderOptions
         /// </example>
         public Func<SecretListEntry, string, string> KeyGenerator { get; set; } = (secret, key) => key;
 
+        /// <summary>
+        /// Defines a function that can be used to customize the <see cref="GetSecretValueRequest"/> before it it sent.
+        /// </summary>
+        /// <example>
+        /// <code>
+        /// GetSecretValueRequestGenerator = secret => new GetSecretValueRequest { SecretId = secret.ARN, VersionStage = "AWSCURRENT" };
+        /// </code>
+        /// </example>
+        public Func<SecretListEntry, GetSecretValueRequest> GetSecretValueRequestGenerator { get; set; } =
```

Comment:
I'm not sure if passing a `SecretListEntry` directly is future-proof enough. Maybe we should create a type that picks the important properties from the SLE so that, in case of need, we can extend it.

Reply:
Added to the README.

Added a `GetSecretValueRequestContext` that contains the secret ARN, Name, and its `VersionsToStages`. Since that's the only thing you can really specify in the request beside the ARN (`GetSecretValueRequest.VersionId` and `GetSecretValueRequest.VersionStage`).

If you request any further changes, please let me know if you prefer I keep everything in one commit or if you plan on squash merging.